includes/class-affiliatewp-leaderboard.php

```diff
@@ -215,6 +215,13 @@ public function affiliate_leaderboard( $atts, $content = null ) {
 			'affiliate_leaderboard'
 		);
 
+        // Sanitize args.
+        $atts['earnings']  = sanitize_text_field( wp_unslash( $atts['earnings'] ) );
+        $atts['referrals'] = sanitize_text_field( wp_unslash( $atts['referrals'] ) );
+        $atts['visits']    = sanitize_text_field( wp_unslash( $atts['visits'] ) );
+        $atts['number']    = absint( $atts['number'] );
+        $atts['orderby']   = sanitize_sql_orderby( $atts['orderby'] );
```

Comment:
```suggestion
$atts = array(
	'earnings'  => sanitize_text_field( wp_unslash( $atts['earnings'] ) ),
	'referrals' => sanitize_text_field( wp_unslash( $atts['referrals'] ) ),
	'visits'    => sanitize_text_field( wp_unslash( $atts['visits'] ) ),
	'number'    => absint( $atts['number'] ),
	'orderby'   => sanitize_sql_orderby( $atts['orderby'] ),
);
```

Why not just assign the array values to ensure nothing weird slips in? Or use `array_merge`?